src/js/messaging/containers/Compose.jsx

```diff
@@ -98,63 +128,49 @@ export class Compose extends React.Component {
+              cssClass="msg-from"
               lastName={message.sender.lastName}
               firstName={message.sender.firstName}
               middleName={message.sender.middleName}/>
           <MessageRecipient
-              errorMessage={composeMessage.errors.recipient}
-              cssClass="messaging-recipient"
+              errorMessage={validations.isValidRecipient(message.recipient) ? '' : composeMessage.errors.recipient}
+              cssClass="msg-recipient msg-field"
               onValueChange={this.props.setMessageField}
               options={this.props.recipients}
               recipient={message.recipient}/>
-          <fieldset className="messaging-subject-field">
-            <legend>Subject line:</legend>
-            <div className="messaging-subject-group">
-              <div>
-                <MessageCategory
-                    categories={messageCategories}
-                    cssClass="messaging-category"
-                    errorMessage={composeMessage.errors.category}
-                    onValueChange={this.props.setMessageField}
-                    category={message.category}/>
-                <MessageSubject
-                    charMax={composeMessage.maxChars.subject}
-                    cssClass="messaging-subject"
-                    errorMessage={composeMessage.errors.subject}
-                    onValueChange={this.props.setMessageField}
-                    placeholder={composeMessage.placeholders.subject}
-                    required={subjectRequired}
-                    subject={message.subject}/>
-              </div>
-            </div>
-          </fieldset>
-          <div className="messaging-write-group">
-            <MessageWrite
-                cssClass="messaging-write"
-                onValueChange={this.handleMessageChange}
-                placeholder={composeMessage.placeholders.message}
-                text={message.text}/>
-            <MessageAttachments
-                hidden={!this.props.message.attachments.length}
-                files={this.props.message.attachments}
-                onClose={this.props.deleteComposeAttachment}/>
-          </div>
-          <MessageSend
+          <MessageSubjectGroup
+              categories={messageCategories}
+              category={message.category}
+              charMax={composeMessage.maxChars.subject}
+              cssErrorClass={subjectError.type ? `msg-compose-error--${subjectError.type}` : undefined}
+              errorMessage={subjectError.hasError ? composeMessage.errors.subjectLine[subjectError.type] : undefined}
+              onCategoryChange={this.props.setMessageField}
+              onSubjectChange={this.props.setMessageField}
+              subject={message.subject}
+              subjectPlaceholder={composeMessage.placeholders.subject}/>
+          <MessageWriteGroup
               allowedMimeTypes={allowedMimeTypes}
+              attachmentsVisible={!this.props.message.attachments.length}
```

Comment:
In the same vein of the above comment about paring down props, this prop can be derived from the `files` prop since they're both based on the `this.props.message.attachments` being passed.